### DISCUSSION TITLE
Repositories URLs configured with gradle ext properties don't work

### DISCUSSION THREAD

**Discussion by mhabzda** [renovatebot/renovate#12975] [Request Help] [answered]
I have the following configuration for a Gradle project
```
allprojects {
    repositories {
        maven { url projectRepositories.google }
        ...
     }
}

ext.projectRepositories = [:]

if (System.getenv().containsKey("CI")) {
    projectRepositories.google = /proxy internal repo/
    ...
}
else {
    projectRepositories.google = "https://dl.google.com/android/maven2/"
    ....
}
```

and it doesn't work with Renovate. It doesn't look up the google repo for my dependencies.

But if I use
```
allprojects {
    repositories {
        google()
        ...
     }
}
```
it works just fine.

I realized that the reason is the shallow script for the Gradle package manager. And because of that, I used `deepExtract` flag and set it to true. But the result was Renovate not finding any Gradle related packages files, my onboarding PR looked like that
![image](https://user-images.githubusercontent.com/26122834/144852706-8125d67c-67d1-4133-bf3b-6561e0992a53.png)

With the usage of `google()` all Gradle-related files were listed in this PR.

I think it shouldn't matter but I used a self-hosted docker configuration on the Gitlab CI.

Is there already a solution for my problem or maybe I should create a feature request?

**Comment by rarkins** [comment, marked as answer]
If you can create a minimal reproduction repo and link to it from a feature request, we can try to debug and address it. BTW please reproduce it with the default (non-deep) approach with minimal dependencies and config.

**Reply by mhabzda** [reply]
I created a future request with reproduction repo linked - https://github.com/renovatebot/renovate/issues/12988
